pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -289,15 +290,33 @@ func removeDuplicateMaterials(mats []common.ProvenanceMaterial) ([]common.Proven
 }
 
 // FromPipelineParamsAndResults extracts type hinted params and results and adds the url and digest to materials.
-func FromPipelineParamsAndResults(ctx context.Context, pro *objects.PipelineRunObject) []common.ProvenanceMaterial {
+func FromPipelineParamsAndResults(ctx context.Context, pro *objects.PipelineRunObject, slsaconfig *slsaconfig.SlsaConfig) []common.ProvenanceMaterial {
 	mats := []common.ProvenanceMaterial{}
 	sms := artifacts.RetrieveMaterialsFromStructuredResults(ctx, pro, artifacts.ArtifactsInputsResultName)
 	mats = append(mats, sms...)
 
 	var commit, url string
-	// search status.PipelineSpec.params
-	if pro.Status.PipelineSpec != nil {
-		for _, p := range pro.Status.PipelineSpec.Params {
+
+	pSpec := pro.Status.PipelineSpec
+	if pSpec != nil {
+		// search type hinting param/results from each individual taskruns
+		if slsaconfig.DeepInspectionEnabled {
+			logger := logging.FromContext(ctx)
+			pipelineTasks := append(pSpec.Tasks, pSpec.Finally...)
+			for _, t := range pipelineTasks {
+				tr := pro.GetTaskRunFromTask(t.Name)
+				// Ignore Tasks that did not execute during the PipelineRun.
+				if tr == nil || tr.Status.CompletionTime == nil {
+					logger.Infof("taskrun status not found for task %s", t.Name)
```

Comment:
This log message is probably not very helpful, and it's maybe a little misleading?

Reply:
Changed. PTAL. Thanks!